### PR TITLE
Feat: lesson24

### PR DESCRIPTION
**Pull request by yuka830**
# [Lesson24](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#24)
```
仕様

- バリデーションはここではなし
- ユーザー名、メールアドレス、パスワードの入力欄と利用規約に関するチェックボックス(画像参照)がある。
- 送信ボタンがあるが振るまいの実装はしないで良い
- 利用規約のテキストを押すと、モーダルが立ち上がり(前回作ったもので良い)、ダミーの利用規約がテキストとして読める。スクロールが一番下に行ったらチェックボックスはcheckedになる。もし開いてもスクロールが下まで行っていなければcheckedはfalseのまま
- checkedがtrueの場合送信ボタンを押下すると別ページのregister-done.htmlに飛ぶ
- register-done.htmlは画面のようなテキストになっている(画面は適当で、遷移できていることが分かれば良い。CSSも書かないでも良い)
```
## [CodeSandbox](https://codesandbox.io/s/lesson24-e7qyo?file=/index.html)
Last commit: [ae01527](https://github.com/yuka830/js-lessons/pull/29/commits/ae01527edb51939129947d1ef50af586ea3f9f6f)

In CodeSandbox
- I changed html file's name from `register.html` to `index.html` because the first one didn't read.
~~The `register-done.html` is displayed with garbled characters(I don't know why), but it is displayed fine locally~~ Now this is fixed, thanks to @nyattsu72

### CODE REVIEW

**Comment by yuka830** [comment]
https://user-images.githubusercontent.com/66966699/149871523-656794d9-ccb3-4dd7-a1e8-e95cde03132c.mov

File: lesson24/register-done.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<h1>会員登録が完了しました</h1>
+<p>登録いただいたメールアドレス宛に登録内容をお送りしましたのでログインして下さい。</p>
```
Comment (nyattsu72) [inline lesson24/register-done.html:13]: It's garbled, I think it is because the file of register-done.html does not declare! DOCTYPE HTML.

Comment (yuka830) [reply lesson24/register-done.html:13]: Wow it's a terrible mistake and an embarrassment... 😭 Thank you so much for telling me!!
[e2b40c8](https://github.com/yuka830/js-lessons/pull/29/commits/e2b40c89561a7600626e24eb6256afed13f4b437)

File: lesson24/register.html [hunk 2]
```diff
@@ -8,35 +8,217 @@
...
+        </div>
+      </dl>
+      <div class="register-form__terms">
+        <input type="checkbox" name="check" id="js-checkBox" disabled />
+        <label for="js-checkBox">
+          <a href="#" id="js-openModal" class="terms-link">利用規約</a>を読み同意しました
+        </label>
+      </div>
+      <div class="register-form__btn">
+        <input type="submit" value="送信" id="js-submitBtn" class="submit-btn btn"></input>
```
Comment (haru-programming) [inline lesson24/register.html]: I'd like to inform you that [Nu Html Checker](https://validator.w3.org/#validate_by_input) was giving me an error due to a single closing tag in the html <input>.

Sorry I can't review the js.
The BEM classing was so beautiful and informative!

**Review by nyattsu72 — APPROVED** [review]

File: lesson24/register.html (hunk 2 above)
Comment (yuka830) [reply lesson24/register.html]: Thank you for telling me about this web site. I checked my coding with it and fixed.
It really helped! I appreciated you :)
[ef300f0](https://github.com/yuka830/js-lessons/pull/29/commits/ef300f0d03b251b2a439b189e3165c964f4a7d12)

**Review by haru-programming — APPROVED** [review]

File: lesson24/main.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  false
+);
+
+const observer = new IntersectionObserver((targets) => {
+  if (targets[0].intersectionRatio === 1) {
+    checkBox.disabled = false;
+    checkBox.checked = true;
+  }
+});
```
Comment (kenmori) [inline lesson24/main.js:27]: 👍

File: lesson24/register.html [hunk 4]
```diff
@@ -216,7 +216,7 @@ <h2 class="sub-title">第15条（準拠法・裁判管轄）</h2>
         <p>以上</p>
       </div>
       <div class="modal-btn">
-        <button id="js-closeModalBtn" class="check-btn btn">確認しました</button>
+        <button id="js-closeModalBtn" class="close-btn btn">閉じる</button>
```
Comment (ghost) [inline lesson24/register.html:219]: Nice!

File: lesson24/register-done.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (ghost) [inline lesson24/register-done.html:2]: NITS
I'm sorry for going into detail.
Please include`register.html` as well.
```suggestion
<html lang="ja">
```

**Review by ghost — APPROVED** [review]

**Pull request merged**